### PR TITLE
Use apptest library and refactor tests

### PR DESCRIPTION
**Pull request by tomahawk28**
Towards giantswarm/giantswarm#13206

### CODE REVIEW

**Comment by tomahawk28** [comment]
```2020-11-27T13:25:08.836714205Z stdout F {"caller":"github.com/giantswarm/chart-operator/v2/service/controller/chart/resource/release/create.go:130","controller":"chart-operator-chart","event":"update","function":"ApplyCreateChange","level":"debug","loop":"9","message":"helm release `external-dns-app` failed","object":"/apis/application.giantswarm.io/v1alpha1/namespaces/giantswarm/charts/external-dns-app","resource":"release","stack":{"annotation":"rendered manifests contain a resource that already exists. Unable to continue with install: PriorityClass \"giantswarm-critical\" in namespace \"\" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key \"app.kubernetes.io/managed-by\": must be set to \"Helm\"; annotation validation error: missing key \"meta.helm.sh/release-name\": must be set to \"external-dns-app\"; annotation validation error: missing key \"meta.helm.sh/release-namespace\": must be set to \"kube-system\"","kind":"unknown","stack":[{"file":"/go/pkg/mod/github.com/giantswarm/helmclient/v2@v2.1.4/pkg/helmclient/install.go","line":52},```

File: integration/templates/external_dns_values.go [hunk 1]
```diff
@@ -3,5 +3,5 @@
 package templates
 
 // ExternalDNSValues values used for external-dns-app in integration test
-const ExternalDNSValues = `e2e: true
```
Comment (tomahawk28) [inline integration/templates/external_dns_values.go:6]: `e2e` value is unnecessary since `priorityClass` is created from apptestctl directly.

Comment (rossf7) [reply integration/templates/external_dns_values.go:6]: Thanks for removing. I think its better to create the priorityclass with apptestctl and then we can simplify the chart.

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**